Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Undo/PrefabUndoUpdateLinkBase.cpp

```diff
@@ -0,0 +1,113 @@
+        }
+
+        void PrefabUndoUpdateLinkBase::Undo()
+        {
+            UpdateLink(m_undoPatch);
+        }
+
+        void PrefabUndoUpdateLinkBase::Redo()
+        {
+            UpdateLink(m_redoPatch);
+        }
+
+        void PrefabUndoUpdateLinkBase::SetLink(LinkId linkId)
+        {
+            m_link = m_prefabSystemComponentInterface->FindLink(linkId);
+            AZ_Assert(m_link.has_value(),
+                "PrefabUndoUpdateLinkBase::SetLink - Link with id '%llu' not found",
+                static_cast<AZ::u64>(linkId));
+            m_templateId = m_link->get().GetTargetTemplateId();
+        }
+
+        void PrefabUndoUpdateLinkBase::GenerateUndoUpdateLinkPatches(const PrefabDom& linkedInstancePatch)
+        {
+            AZ_Assert(m_link.has_value(),
+                "PrefabUndoUpdateLinkBase::GenerateUndoUpdateLinkPatches - m_link is still invalid");
+
+            //Cache current link DOM for undo link update.
+            m_undoPatch.CopyFrom(m_link->get().GetLinkDom(), m_undoPatch.GetAllocator());
+
+            //Get DOM of the link's source template.
+            TemplateId sourceTemplateId = m_link->get().GetSourceTemplateId();
+            TemplateReference sourceTemplate = m_prefabSystemComponentInterface->FindTemplate(sourceTemplateId);
+            AZ_Assert(sourceTemplate.has_value(),
+                "PrefabUndoUpdateLinkBase::GenerateUndoUpdateLinkPatches - Source template not found");
+            PrefabDom& sourceDom = sourceTemplate->get().GetPrefabDom();
+
+            //Get DOM of instance that the link points to.
+            PrefabDomValueReference instanceDomRef = m_link->get().GetLinkedInstanceDom();
+            AZ_Assert(instanceDomRef.has_value(),
+                "PrefabUndoUpdateLinkBase::GenerateUndoUpdateLinkPatches -Linked Instance DOM not found");
+            PrefabDom instanceDom;
+            instanceDom.CopyFrom(instanceDomRef->get(), instanceDom.GetAllocator());
+
+            //Apply patch for adding new entity under parent entity to the instance.
+            [[maybe_unused]] AZ::JsonSerializationResult::ResultCode result = PrefabDomUtils::ApplyPatches(
+                instanceDom, instanceDom.GetAllocator(), linkedInstancePatch);
+            AZ_Warning(
+                "Prefab",
+                (result.GetOutcome() != AZ::JsonSerializationResult::Outcomes::Skipped) &&
+                (result.GetOutcome() != AZ::JsonSerializationResult::Outcomes::PartialSkip),
+                "Some of the patches are not successfully applied:\n%s.", PrefabDomUtils::PrefabDomValueToString(m_redoPatch).c_str());
+
+            // Remove the link ids if present in the DOMs. We don't want any overrides to be created on top of linkIds because
+            // linkIds are not persistent and will be created dynamically when prefabs are loaded into the editor.
+            instanceDom.RemoveMember(PrefabDomUtils::LinkIdName);
+            sourceDom.RemoveMember(PrefabDomUtils::LinkIdName);
```

Comment:
Turns out the two dom do contains LinkId key in them at very beginning. I will make it a AZ_Warning here since assert causes unit tests failed directly.